WcaOnRails/lib/city_validator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class CityValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if record.country.nil? || value.blank?
+
+    city_validator = self.class.get_validator_for_country(record.country.iso2)
+    if city_validator
+      reason_why_invalid = city_validator.reason_why_invalid(value)
+      record.errors.add(attribute, reason_why_invalid) if reason_why_invalid
+    end
+  end
+
+  def self.add_validator_for_country(country_iso2, validator)
+    @validators_by_country_iso2 ||= {}
```

Comment:
Doesn't matter here, but what about just initializing this class instance variable with `{}`? What I mean is `@validators_by_country_iso2 = {}` outside the methods. Theoretically you could call `self.get_validator_for_country` before `self.add_validator_for_country`, which would result in an error now.

Reply:
Oh, good call! That's definitely better.